frontend/src/services/set-validation-language.ts

```diff
@@ -0,0 +1,19 @@
+
+import {Language, setLanguage} from "zero-zummon"
+
+export function setValidationLanguage(): void {
```

Comment:
Reduce the code repetition.
Move language logic to one place
a map with what we currently support, easier to maintain and extend if additional languages are supported

import { Language, setLanguage as kotlinSetLanguage } from "zero-zummon";

const setPreferredLanguage = () => {
    const languageMap = {
        en: Language.en,
        nl: Language.nl
    };

    for (const language of navigator.languages) {
        const languageCode = language.split('-')[0]; // Extract primary language subtag
        if (languageMap[languageCode]) { // It no matching language, fallback in the default.
            kotlinSetLanguage(languageMap[languageCode]);
            return;
        }
    }
};

Reply:
For just 2 languages this seems less understandable.